fuse/data/tokenizers/modular_tokenizer/inject_utils.py

```diff
@@ -147,10 +144,12 @@ def prepare_info_for_model_step(
 
 
         """
-        scalars_indices = []
-        scalars_values = []
-        scalars_masked_indices = []
-        prev_index_end = -1
+        ## both `all_scalars_values` and `all_scalars_valid_mask` will contain torch tensors, which will be concatanated in the end of this function
+        all_scalars_values = (
+            []
+        )  # one scalar for every element, `scalar_default_unfound_value` is used for elements that aren't scalars
+        all_scalars_valid_mask = []  # for each element, whether it's a scalar or not
+        scalar_default_unfound_value = -1000.0
```

Comment:
What does it mean?

Reply:
since we now keep the scalars values and mask at the size of the entire sequence, this is the default value for places that don't actually have a scalar value.
I chose -1000 and not something like 0 to make sure it pops up easily if there are mistakes down the road